sound/soc/sof/intel/hda-loader.c

```diff
@@ -470,3 +471,122 @@ int hda_dsp_post_fw_run(struct snd_sof_dev *sdev)
 	/* re-enable clock gating and power gating */
 	return hda_dsp_ctrl_clock_power_gating(sdev, true);
 }
+
+static int hda_dsp_ext_man_get_cavs_config_data(struct snd_sof_dev *sdev,
+				   const struct sof_ext_man_elem_header *hdr)
+{
+	const struct sof_ext_man_cavs_config_data *config_data =
+		container_of(hdr, struct sof_ext_man_cavs_config_data, hdr);
+	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
+	int i;
+
+	for (i = 0; i < SOF_EXT_MAN_CAVS_CONFIG_LAST_ELEM - 1; i++)
```

Comment:
why not SOF_EXT_MAN_CAVS_CONFIG_LAST_ELEM = SOF_EXT_MAN_CAVS_CONFIG_CAVS_LPRO? then your loop condition becomes simple (for i = 0; i < SOF_EXT_MAN_CAVS_CONFIG_LAST_ELEM;i ++)
I find it odd that you have < and -1 in the check. Its correct but just not intuitive

Reply:
this is good idea.

Reply:
Using any statically defined enum value in for loop end condition leads to highly vulnerable solution, let thing what's going to  happen after adding any new enym element to this list and loading old FW (without this new element) - you will read memory out of element definition.